File: ball.lua
Ball = {style = "", xPos = 0, yPos = 0, radius = 0, segments = 20, speed = 0, xVelocity = 0, yVelocity = 0}

function Ball:new (o)
    o = o or {}
    setmetatable(o, self)
    self.__index = self
    return o
end

function Ball:reset(centerX, centerY)
    self.xVelocity = 0
    self.yVelocity = 0
    self.xPos = centerX
    self.yPos = centerY
end

function Ball:draw()
    love.graphics.circle(self.style, self.xPos, self.yPos, self.radius, self.segments)
end

File: C#-Monogame/GameObjects/Ball.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

public class Ball
{
    public Texture2D _texture;
    public Vector2 pos;
    public Color _color;
    public float speed;
    public float xVelocity = 0;
    public float yVelocity = 0;

    public Ball(Vector2 pos, float speed, Color _color)
    {
        this.pos = pos;
        this.speed = speed;
        this._color = _color;
    }

    public void UpdatePosition()
    {
        this.pos.X += xVelocity;
        this.pos.Y += yVelocity;
    }

    public int GetRadius()
    {
        return _texture.Width / 2;
    }

    public void Reset(int centerX, int centerY)
    {
        this.pos.X = centerX;
        this.pos.Y = centerY;
        this.xVelocity = 0;
        this.yVelocity = 0;
    }
}